### PR TITLE
gh-97696 Add documentation for get_coro() behavior with eager tasks

### PR DESCRIPTION
**Pull request by jbower-fb**
This PR caught the plague and had to be killed with fire. A new one was opened to replace it https://github.com/python/cpython/pull/104304.

In response to comments on this issue.

* Issue: gh-97696

----
:books: Documentation preview :books:: https://cpython-previews--104189.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/asyncio-task.rst [hunk 1]
```diff
@@ -1153,6 +1153,11 @@ Task Object
 
       Return the coroutine object wrapped by the :class:`Task`.
 
+      .. note::
+
```
Comment (gaogaotiantian) [inline Doc/library/asyncio-task.rst]: A trailing space here :)

**Review by gvanrossum — COMMENTED** [review]
LG, but let's wait until Kumar's feedback has been resolved.

File: Doc/library/asyncio-task.rst [hunk 2]
```diff
@@ -1153,6 +1153,11 @@ Task Object
 
       Return the coroutine object wrapped by the :class:`Task`.
 
+      .. note::
```
Comment (kumaraditya303) [inline Doc/library/asyncio-task.rst:1179]: Add a versionchanged directive.

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: Doc/library/asyncio-task.rst [hunk 3]
```diff
@@ -1153,6 +1153,11 @@ Task Object
...
+
+        This will return ``None`` for Tasks which have already
+        completed eagerly.
```
Comment (gvanrossum) [inline Doc/library/asyncio-task.rst]: I feel this requires a link to the eager task factory stuff.

Comment (kumaraditya303) [reply Doc/library/asyncio-task.rst]: Good idea.

**Comment by jbower-fb** [comment]
Apologies for the spam everyone. Looks like I did a rebase where I should have done a merge. I'm going to close this PR and open a new one to clear things for people.

File: Doc/library/asyncio-task.rst (hunk 3 above)
Comment (jbower-fb) [reply Doc/library/asyncio-task.rst]: Resolved in the new PR https://github.com/python/cpython/pull/104304

File: Doc/library/asyncio-task.rst (hunk 2 above)
Comment (jbower-fb) [reply Doc/library/asyncio-task.rst:1179]: Resolved in the new PR https://github.com/python/cpython/pull/104304.

**Pull request closed without merging**